src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Ball {
+    private final String color;
+    private final int number;
+
+    public enum Color {
+        RED,
+        GREEN,
+        BLUE,
+        YELLOW,
+        ORANGE,
+        PURPLE,
+        BLACK,
+        WHITE
+    }
+
+    public Ball(String color, int number) {
+        this.color = color;
```

Comment:
The constructor parameter 'color' should be of type 'Color' (the enum defined within the class), not 'String'. This ensures that only the predefined colors are used when creating a Ball instance.